setup.py

```diff
@@ -110,6 +110,27 @@
   DEFINE_MACROS += (('PyMODINIT_FUNC', pymodinit),)
 
 
+# By default, Python3 distutils enforces compatibility of
+# plugins with the OSX version Python3 was built with.
+# For Python3.4, this is OSX 10.6, but we need Thread Local Support (__thread)
+# from OSX 10.7
+def get_os_version_osx():
+  import sysconfig, re
+  platform = sysconfig.get_platform()
+  platform_match = re.match("macosx-(.*)-.*", platform)
+  if platform_match:
+    if platform_match.groups()[0] <= "10.6":
+      return "10.7"
+    else:
+      return platform_match.groups()[0]
+  else:
+    raise Exception("No match for REGEX in get_os_version_osx")
+
+
+if "darwin" in sys.platform and PY3:
+  os.environ['MACOSX_DEPLOYMENT_TARGET']=get_os_version_osx()
```

Comment:
I would prefer to set ```MACOSX_DEPLOYMENT_TARGET``` in the minimum number of cases given the headaches that it has already caused us.  Maybe in the if statement we can ensure we are compiling against a version 10.6 or less.